core/src/test/java/kafka/test/ClusterConfig.java

```diff
@@ -67,13 +69,16 @@ public class ClusterConfig {
         this.listenerName = listenerName;
         this.trustStoreFile = trustStoreFile;
         this.metadataVersion = metadataVersion;
-        this.serverProperties = copyOf(serverProperties);
-        this.producerProperties = copyOf(producerProperties);
-        this.consumerProperties = copyOf(consumerProperties);
-        this.adminClientProperties = copyOf(adminClientProperties);
-        this.saslServerProperties = copyOf(saslServerProperties);
-        this.saslClientProperties = copyOf(saslClientProperties);
-        perBrokerOverrideProperties.forEach((brokerId, props) -> this.perBrokerOverrideProperties.put(brokerId, copyOf(props)));
+        this.serverProperties = Collections.unmodifiableMap(serverProperties);
+        this.producerProperties = Collections.unmodifiableMap(producerProperties);
+        this.consumerProperties = Collections.unmodifiableMap(consumerProperties);
+        this.adminClientProperties = Collections.unmodifiableMap(adminClientProperties);
+        this.saslServerProperties = Collections.unmodifiableMap(saslServerProperties);
+        this.saslClientProperties = Collections.unmodifiableMap(saslClientProperties);
+        this.perBrokerOverrideProperties = Collections.unmodifiableMap(
```

Comment:
Pardon me, I'm not quite clear about this comment. Could you explain more ? :smiley: 
For convenience, this pr add something like `public Builder putServerProperty(String key, String value) ` in each configurations like `serverProperties`, `consumerProperties`. So we don't need to do deep copy in ClusterConfig.Builder. 
Or did you mean we need to use `setServerProperty(Map<String, String> serverProperties)`  instead of `putServerProperty(String key, String value)`  ?

Reply:
The purpose of this PR is to make `ClusterConfig` be immutable. However, `ClusterConfig` have only immutable "view" of `serverProperties`, `producerProperties`, etc. It means `ClusterConfig` is still mutable object since callers can change inner variables of `ClusterConfig` by updating `ClusterConfig#Builder`. For example, user can call `putSaslServerProperty` to chagne `saslServerProperties` of `ClusterConfig`

Reply:
Got it, thanks for explanation.